Tests/RemoteMessagingTests/Matchers/CommonAppAttributeMatcherTests.swift

```diff
@@ -26,6 +26,7 @@ import XCTest
 class CommonAppAttributeMatcherTests: XCTestCase {
 
     private var matcher: CommonAppAttributeMatcher!
+    private let versionNumber = "3.2.1"
```

Comment:
I've moved this to using a fixed version number in the test suite, since `AppVersion.shared.versionNumber` in macOS BSK tests returns the OS version, which is 2 components for me, which causes some problems in the tests.

Also, the approach of using a dynamic version number led to some weird test cases, like tests which decrement a component would produce version strings like `16.0.-1`.